llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -1018,9 +1018,10 @@ static bool isAddressUse(const TargetTransformInfo &TTI,
         isAddress = true;
       break;
     default: {
-      MemIntrinsicInfo IntrInfo;
-      if (TTI.getTgtMemIntrinsic(II, IntrInfo)) {
-        if (IntrInfo.PtrVal == OperandVal)
+      std::pair<MemIntrinsicInfo, SmallVector<InterestingMemoryOperand, 1>>
+          MemInfo = TTI.getTgtMemIntrinsic(II);
+      if (MemInfo.first.PtrVal != nullptr) {
```

Comment:
Is OperandVal always non-null? If so, it could never match a null `MemInfo.first.PtrVal`.

If we do this check, then use `&&` to join the two ifs instead of nesting them.